sdk/trace/provider.go

```diff
@@ -99,6 +102,14 @@ func NewTracerProvider(opts ...TracerProviderOption) *TracerProvider {
 	return tp
 }
 
+// SetClock supports setting a mock clock for testing or some clock with custom
+// time correction logic.
+func (p *TracerProvider) SetClock(clock tracetime.Clock) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.clock = clock
+}
```

Comment:
is getting the clock synchronized?

Reply:
Resolved per https://github.com/open-telemetry/opentelemetry-go/pull/2052/files#r664118002